File: Framework/EngineSupport/Unity/MMIUnity.TargetEngine/MMIUnity.TargetEngine/MMISettings.cs
// SPDX-License-Identifier: MIT
// The content of this file has been developed in the context of the MOSIM research project.
// Original author(s): Felix Gaisbauer

using UnityEngine;

namespace MMIUnity.TargetEngine
{
    /// <summary>
    /// Class which contains all MMI settings
    /// </summary>
    public class MMISettings : MonoBehaviour
    {
        /// <summary>
        /// The port for the access
        /// </summary>
        [Header("The port of the MMIRegister")]
        public int MMIRegisterPort = 9009;


        [Header("The address of the MMI Register")]
        public string MMIRegisterAddress = "127.0.0.1";

        /// <summary>
        /// The port for the access
        /// </summary>
        [Header("The port used for remotely accessing the scene")]
        public int RemoteSceneAccessPort = 9000;

        /// <summary>
        /// The port for the access
        /// </summary>
        [Header("The address used for remotely accessing the scene")]
        public string RemoteSceneAccessAddress = "127.0.0.1";

        /// <summary>
        /// The port for the external write access
        /// </summary>
        [Header("The port used for remotely manipulating the scene")]
        public int RemoteSceneWritePort = 9001;


        /// <summary>
        /// The port for the access
        /// </summary>
        [Header("The address used for remotely manipulating the scene")]
        public string RemoteSceneWriteAddress = "127.0.0.1";

        /// <summary>
        /// Specifies whether the scene is accessible for external clients via thrift server
        /// </summary>
        [Header("Specifies whether the scene is accessible for external clients via thrift server")]
        public bool AllowRemoteSceneConnections = true;

        
        /// <summary>
        /// The port for the skeleton access
        /// </summary>
        [Header("The port used for remotely accessing the skeleton")]
        public int RemoteSkeletonAccessPort = 9999;


        /// <summary>
        /// The port for the skeleton access
        /// </summary>
        [Header("The address used for remotely accessing the skeleton")]
        public string RemoteSkeletonAccessAddress = "127.0.0.1";



        /// <summary>
        /// Specifies whether the skeleton is accessible for external clients via thrift server
        /// </summary>
        [Header("Specifies whether the skeleton is accessible for external clients via thrift server")]
        public bool AllowRemoteSkeletonConnections = true;


    }


}

File: Framework/EngineSupport/Unreal/MOSIMPlugin/Source/MMMIScene/Private/MMISettings.cpp
// SPDX-License-Identifier: MIT
// The content of this file has been developed in the context of the MOSIM research project.
// Original author(s): Stephan Adam

// Storage for the most important IP and Port settings of the Plugin

#include "MMISettings.h"

// constructor
MMISettings::MMISettings()
    : MMIRegisterPort( 9009 ),
      MMIRegisterAddress( "127.0.0.1" ),
      MIPRegisterAddress( MIPAddress{} ),

      RemoteSceneAccessPort( 9000 ),
      RemoteSceneAccessAddress( "127.0.0.1" ),
      MIPRemoteSceneAccessAddress( MIPAddress{} ),

      RemoteSceneWritePort( 9001 ),
      RemoteSceneWriteAddress( "127.0.0.1" ),
      MIPRemoteSceneWriteAddress( MIPAddress{} ),
      AllowRemoteSceneConnections( true ),

      // TODO: probably unrequired
      // RemoteSkeletonAccessPort(9999),
      // RemoteSkeletonAccessAddress("127.0.0.1"),
      // MIPRemoteSkeletonAccessAddress(MIPAddress{}),
      // AllowRemoteSkeletonConnections(true)

      RemoteCoSimulationAccessPort( 9011 ),
      RemoteCoSimulationAccessAddress( "127.0.0.1" ),
      MIPRemoteCoSimulationAccessAddress( MIPAddress{} )

{
    MIPRegisterAddress.__set_Address( this->MMIRegisterAddress );
    MIPRegisterAddress.__set_Port( this->MMIRegisterPort );

    MIPRemoteSceneAccessAddress.__set_Address( this->RemoteSceneAccessAddress );
    MIPRemoteSceneAccessAddress.__set_Port( this->RemoteSceneAccessPort );

    MIPRemoteSceneWriteAddress.__set_Address( this->RemoteSceneWriteAddress );
    MIPRemoteSceneWriteAddress.__set_Port( this->RemoteSceneWritePort );

    MIPRemoteCoSimulationAccessAddress.__set_Address( this->RemoteCoSimulationAccessAddress );
    MIPRemoteCoSimulationAccessAddress.__set_Port( this->RemoteCoSimulationAccessPort );

    // TODO: probably unrequired
    // MIPRemoteSkeletonAccessAddress.__set_Address(this->RemoteSkeletonAccessAddress);
    // MIPRemoteSkeletonAccessAddress.__set_Port(this->RemoteSkeletonAccessPort);
}
